src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -18,18 +18,6 @@ namespace AppInstaller::CLI::Workflow
     using namespace AppInstaller::Manifest;
     using namespace AppInstaller::Repository;
 
-    void EnsureMinOSVersion(Execution::Context& context)
-    {
-        const auto& manifest = context.Get<Execution::Data::Manifest>();
-
-        if (!manifest.MinOSVersion.empty() &&
-            !Runtime::IsCurrentOSVersionGreaterThanOrEqual(Version(manifest.MinOSVersion)))
-        {
-            context.Reporter.Error() << Resource::String::InstallationRequiresHigherWindows << ' ' << manifest.MinOSVersion << std::endl;
-            AICLI_TERMINATE_CONTEXT(HRESULT_FROM_WIN32(ERROR_OLD_WIN_VERSION));

```

Comment:
I understand the motivation to remove this, but it does mean that we will not message the reason for failure to the user (or ourselves for the failure code).  Maybe it is upcoming in my reading of the PR, but we need a way to message the failure state of no applicable installers.

Reply:
Yes, the MinOSVersion check is moved to installer selection logic since in the new world each installer has its own MinOSVersion. It will be part of the No_Applicable_Installer error

Reply:
But I think we need to do a better job of messaging the reason that there are not applicable installers.  Some of that burden could be placed on search to filter them out before we even attempt to install, but I prefer to show everything and have the user be informed that we don't have their flavor of installer rather than we don't even know what they are talking about.

It might be more complex, and thus not piled on to this PR, but I think we should attempt to give some reason as to why there are no applicable installers.  Imagine:
```
for each installer
  reason[i] = GetAllReasonsWhyNotApplicable(installer)
if (AndAllReasons() != 0)
  ReportReasons(AndAllReasons())
```
So if all installers require a newer OS, we say that.  If they are all ARM, we can say that they are for a different processor architecture.  If one required a newer OS and one is ARM, we say both somehow.

---
In reply to: [571311595](https://github.com/microsoft/winget-cli/pull/740#discussion_r571311595) [](ancestors = 571311595)

Reply:
I'll add a todo for better message regarding installer selection

---
In reply to: [571318031](https://github.com/microsoft/winget-cli/pull/740#discussion_r571318031) [](ancestors = 571318031,571311595)